pkg/integrations/azure_exporter/azure_exporter.go

```diff
@@ -0,0 +1,263 @@
+	}
+
+	settings := &metrics.RequestMetricSettings{
+		Subscriptions:   c.Subscriptions,
+		Metrics:         c.Metrics,
+		ResourceType:    c.ResourceType,
+		TagLabels:       c.IncludedResourceTags,
+		Aggregations:    c.MetricAggregations,
+		Filter:          c.ResourceGraphQueryFilter,
+		MetricNamespace: c.MetricNamespace,
+		MetricTemplate:  c.MetricNameTemplate,
+		HelpTemplate:    c.MetricHelpTemplate,
+
+		// Interval controls data aggregation timeframe ie 1 minute or 5 minutes aggregations
+		// Timespan controls query start and end time
+		// Interval == Timespan to ensure we only get a single data point any more than that is useless
+		Timespan: c.Timespan,
+		Interval: to.Ptr[string](c.Timespan),
+
+		// Unused settings just here to capture they are intentionally not set
+		Name:  "",
+		Cache: nil,
+	}
+
+	// Dimensions can only be retrieved via an obscure manner of including a "metric filter" on the query
+	// This isn't documented in the Azure API only the exporter: https://github.com/webdevops/azure-metrics-exporter#virtualnetworkgateway-connections-dimension-support
+	if len(c.IncludedDimensions) > 0 {
+		builder := strings.Builder{}
+		for index, dimension := range c.IncludedDimensions {
+			if _, err := builder.WriteString(dimension); err != nil {
+				return nil, err
+			}
+			if _, err := builder.WriteString(" eq '*'"); err != nil {
+				return nil, err
+			}
+			// Not the last dimension add an `and`
+			if index != (len(c.IncludedDimensions) - 1) {
+				if _, err := builder.WriteString(" and "); err != nil {
+					return nil, err
+				}
+			}
+		}
+		settings.MetricFilter = builder.String()
+
+		// The metric filter introduces a secondary complexity where data is limited by a "top" parameter (default 10)
+		// We don't get any knowledge if the result is cut off and there's no support for paging
+		// set the value as high as possible to hopefully prevent issues
+		settings.MetricTop = to.Ptr[int32](math.MaxInt32)
+		settings.MetricOrderBy = "" // Order is only relevant if top won't return all the results our high value should prevent this
+	}
+
+	concurrencyConfig := azure_config.Opts{
+		Prober: struct {
+			// Limits the number of subscriptions which can concurrently be sending metric requests - value taken from OSS exporter
+			ConcurrencySubscription int `long:"concurrency.subscription"          env:"CONCURRENCY_SUBSCRIPTION"           description:"Concurrent subscription fetches"                                  default:"5"`
+			// Limits the number of concurrent metric requests for a single subscription  - value taken from OSS exporter
+			ConcurrencySubscriptionResource int  `long:"concurrency.subscription.resource" env:"CONCURRENCY_SUBSCRIPTION_RESOURCE"  description:"Concurrent requests per resource (inside subscription requests)"  default:"10"`
+			Cache                           bool `long:"enable-caching"                    env:"ENABLE_CACHING"                     description:"Enable internal caching"`
```

Comment:
the annotation in here come from the azure lib right?

Reply:
Yeah the azure lib defines these as embedded structs so to pass it to the prober I have to inherit this. I changing this in the upstream because I wasn't sure how it would impact the config parsing.